substratevm/src/com.oracle.svm.hosted/src/com/oracle/svm/hosted/phases/SharedGraphBuilderPhase.java

```diff
@@ -211,7 +216,28 @@ protected void handleUnresolvedNewMultiArray(JavaType type, ValueNode[] dims) {
 
         @Override
         protected void handleUnresolvedInstanceOf(JavaType type, ValueNode object) {
+            // The INSTANCEOF byte code refers to a type that could not be resolved.
+            // INSTANCEOF must not throw an exception if the object is null.
+            BeginNode nullObj = graph.add(new BeginNode());
+            BeginNode nonNullObj = graph.add(new BeginNode());
+            append(new IfNode(graph.addOrUniqueWithInputs(IsNullNode.create(object)),
+                            nullObj, nonNullObj, BranchProbabilityNode.NOT_FREQUENT_PROFILE));
+
+            // Case where the object is not null, and type could not be resolved: Throw an
+            // exception.
+            if (lastInstr != null) {
```

Comment:
These
```java
if (lastInstr != null) {
    lastInstr.setNext(X);
}
```
blocks are suspicious, i guess they are just never taken, otherwise they would create a problem in the graph with the begin nodes having 2 predecessors: the `If` and something else.

In the first instance, we just appended an `If`, so the logic in `append`->`updateLastInstruction` will ensure `lastInstr` is null.
In the second instance, we just appended a control flow sink, so the same logic also ensures `lastInstr` is null.

Reply:
Methods that update `lastInstr` are not documented, and usage patterns of `lastInstr` are not clear to me. Should I remove these calls to `setNext()`?

Reply:
I think in general it shouldn't be necessary to update `lastInstr` manually. Methods like `append` should take care of it.
However, as you already found out, when creating some control-flow you'll need to set `lastInstr` to allow other methods to link what they are building with your control-flow structure.

I think the way you set `lastInstr` to the being nodes makes sense.

> Should I remove these calls to `setNext()`?

Yes

Reply:
Done.